vertical-pod-autoscaler/pkg/admission-controller/logic/server.go

```diff
@@ -100,8 +100,8 @@ func (s *AdmissionServer) getPatchesForPodResourceRequest(raw []byte, namespace
 	vpaObservedContainersValue := annotations.GetVpaObservedContainersValue(&pod)
 	patches = append(patches, patchRecord{
 		Op:    "add",
-		Path:  "/metadata/annotations",
-		Value: map[string]string{annotations.VpaObservedContainersLabel: vpaObservedContainersValue},
+		Path:  fmt.Sprintf("/metadata/annotations/%s", annotations.VpaObservedContainersLabel),
```

Comment:
AFAIU this will fail if there are no annotations on the pod. Please either leave a comment on why we know annotations are not empty here or add an if for empty pod.Annotations as above with vpaUpdates annotation.

Reply:
(bonus point for extracting to a helper method :) )

Reply:
Ok. I changed the logic there:
- new we create patch to add empty annotations (if needed)
- we add resource annotation patch (if needed)
- we add observed containers annotation patch 

It creates additional patch compared to previous version, but IMHO the logic seems easier to understand now.

Reply:
Agreed, also harder to add another annotation in a wrong way.